### PR TITLE
Fix `this_fccsw`->`setup.sh`, added printouts

### PR DESCRIPTION
**Pull request by vvolkl**
Implements the suggestion from https://github.com/HEP-FCC/FCCSW/pull/353 to rename `this_fccsw.sh` and uses absolute paths from cmake, to avoid the issue with `dirname`. Also adds two printouts that I find useful, one gives some feedback what `init.sh` is doing, the other one prints the output filename at the end of the job.

### CODE REVIEW

File: FWCore/src/components/PodioOutput.cpp [hunk 1]
```diff
@@ -128,5 +128,6 @@ StatusCode PodioOutput::finalize() {
   m_datatree->Write();
   m_file->Write();
   m_file->Close();
+  info() << "Data written to: " << m_filename << endmsg;
```
Comment (JavierCVilla) [inline FWCore/src/components/PodioOutput.cpp]: Is this related to this PR?

File: CMakeLists.txt [hunk 2]
```diff
@@ -39,19 +39,20 @@ endif()
...
                     WORLD_EXECUTE WORLD_READ
         RENAME run)
-install(FILES ${CMAKE_BINARY_DIR}/this_fccsw.sh
+
+install(FILES ${CMAKE_BINARY_DIR}/setup.sh
         DESTINATION .
-        PERMISSIONS OWNER_EXECUTE OWNER_WRITE OWNER_READ
-                    GROUP_EXECUTE GROUP_READ
-                    WORLD_EXECUTE WORLD_READ)
+        PERMISSIONS OWNER_WRITE OWNER_READ
```
Comment (JavierCVilla) [inline CMakeLists.txt]: Don't we want the script to be executable?

File: cmake/setup.sh.in [hunk 3]
```diff
@@ -1,11 +1,17 @@
...
 
 export PATH=$PATH:$FCCSWBASEDIR/scripts:$FCCSWBASEDIR/bin
 export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$FCCSWBASEDIR/lib
 export PYTHONPATH=$PYTHONPATH:$FCCSWBASEDIR/lib:$FCCSWBASEDIR/python
 export CMAKE_PREFIX_PATH=$CMAKE_PREFIX_PATH:$FCCSWBASEDIR
+
+
+# make sure gaudi scripts and modules are available
+export PYTHONPATH=$PYTHONPATH:$FCCVIEW/lib:$FCCVIEW/python
+export PATH=$PATH:$FCCVIEW/scripts:$FCCVIEW/bin
```
[the remark this answers is not in the document]
Comment (vvolkl) [reply cmake/setup.sh.in]: I agree with you, but for a first step we should make sure the cvmfs installation are working (and for other users who set up their own gaudi (like me :) ) this will probably have no effect as it just appends `/lib` etc to the paths, which won't do anything). But yes, the externals should set up these paths for Gaudi to work properly. Since I cannot setup FCCSW without the Gaudi python modules in the PYTHONPATH, I propose to leave it like this now and delete these lines once the setup script for the fcc view is fixed.

File: CMakeLists.txt (hunk 2 above)
Comment (vvolkl) [reply CMakeLists.txt]: No, we want the script to be sourced. Executing it means running it in its shell which does not have any effect at all. And having it executable can lead to strange behavior like this: https://github.com/HEP-FCC/FCCSW/issues/355

File: cmake/setup.sh.in (hunk 3 above)
[the remark this answers is not in the document]
Comment (JavierCVilla) [reply cmake/setup.sh.in]: In that case I would rather fix the template in the fcc view and patch the current setup.sh installed in cvmfs. I can do that so if you want.

File: FWCore/src/components/PodioOutput.cpp (hunk 1 above)
Comment (vvolkl) [reply FWCore/src/components/PodioOutput.cpp]: No. Do you think it needs its own PR?

File: cmake/setup.sh.in (hunk 3 above)
[the remark this answers is not in the document]
Comment (vvolkl) [reply cmake/setup.sh.in]: Ah great, if you can do that, that would the best solution. Let me know when the patch is applied, then I'll test it again.

[the remark this answers is not in the document]
Comment (JavierCVilla) [reply cmake/setup.sh.in]: I already applied it, can you please give it a try?

File: FWCore/src/components/PodioOutput.cpp (hunk 1 above)
Comment (JavierCVilla) [reply FWCore/src/components/PodioOutput.cpp]: IMO it is better to have one single PR for logical change, but it's fine.

File: cmake/setup.sh.in (hunk 3 above)
[the remark this answers is not in the document]
Comment (vvolkl) [reply cmake/setup.sh.in]: Yes, it's working now, thanks

**Comment by vvolkl** [comment]
I still get these ROOT warnings on I/O:
```
Error in cling::AutoloadingVisitor::InsertIntoAutoloadingState:
   Missing FileEntry for ExampleHit.h
   requested to autoload type ConstExampleHit
Error in cling::AutoloadingVisitor::InsertIntoAutoloadingState:
   Missing FileEntry for ExampleClusterConst.h
   requested to autoload type ConstExampleCluster
Error in cling::AutoloadingVisitor::InsertIntoAutoloadingState:
   Missing FileEntry for ExampleClusterData.h
   requested to autoload type ExampleClusterData
Error in cling::AutoloadingVisitor::InsertIntoAutoloadingState:
   Missing FileEntry for ExampleForCyclicDependency1Data.h
   requested to autoload type ExampleForCyclicDependency1Data
Error in cling::AutoloadingVisitor::InsertIntoAutoloadingState:
   Missing FileEntry for ExampleForCyclicDependency2Const.h


```
will create an issue.
